File: app/src/main/java/uk/co/sundroid/util/LogWrapper.java
package uk.co.sundroid.util;

import uk.co.sundroid.BuildConfig;
import android.util.Log;

public class LogWrapper {
	
	public static void d(String tag, String message) {
		if (BuildConfig.DEBUG) {
			Log.d("sundroid." + tag, message);
		}
	}
	
	public static void i(String tag, String message) {
		if (BuildConfig.DEBUG) {
			Log.i("sundroid." + tag, message);
		}
	}
	
	public static void e(String tag, String message, Throwable tr) {
		if (BuildConfig.DEBUG) {
			Log.e("sundroid." + tag, message, tr);
		}
	}
	
}

File: app/src/main/kotlin/uk/co/sundroid/util/log/LogWrapper.kt
package uk.co.sundroid.util.log

import android.util.Log
import uk.co.sundroid.BuildConfig
import java.util.*

fun d(tag: String, message: String) {
    if (BuildConfig.DEBUG) {
        Log.d("sundroid.$tag", message)
    }
}
fun i(tag: String, message: String) {
    if (BuildConfig.DEBUG) {
        Log.i("sundroid.$tag", message)
    }
}

fun e(tag: String, message: String, tr: Throwable) {
    if (BuildConfig.DEBUG) {
        Log.e("sundroid.$tag", message, tr)
    }
}

fun displayCalendar(calendar: Calendar?): String? {
    if (!BuildConfig.DEBUG) {
        return "Disabled"
    }
    if (calendar == null) {
        return "null"
    }
    val zoneDST = calendar.timeZone.inDaylightTime(Date(calendar.timeInMillis + 12 * 60 * 60 * 1000))
    val zoneName = calendar.timeZone.getDisplayName(zoneDST, TimeZone.LONG)
    val zoneCode = calendar.timeZone.getDisplayName(zoneDST, TimeZone.SHORT)
    return calendar[Calendar.YEAR].toString() + "-" + (calendar[Calendar.MONTH] + 1) + "-" + calendar[Calendar.DAY_OF_MONTH] + " " +
            calendar[Calendar.HOUR_OF_DAY] + ":" + calendar[Calendar.MINUTE] + ":" + calendar[Calendar.SECOND] + " " + zoneName + " (" + zoneCode + ")"
}